Tutorial - Update Link.comments return type (#3547)
dotansimha/graphql-yoga: website/src/pages/tutorial/basic/08-graph-relations.mdx

On the Prisma API level, the method call for loading will always return an array. It does not
distinguish between an empty array and an array with comments.

> Now when we talk about the GraphQL schema, The main question is - how do we choose to represent the
> option of `no comments` in terms of types. What's the best way to represent
> `there's no Comment related to the Link` concept.

Reviewer: ```suggestion
As we design our GraphQL schema we need to consider how we will model the concept of a Link without comments.
```

We have a couple of options:

First in terms of the array itself, we can define it as nullable or not. Then for the elements of
the array, also there we can define them as nullable or not.

> Here are the 4 options, first two are for nullable array and the last two or for none nullable

Reviewer 2: ```suggestion
Here are the 4 options, first two are for nullable array and the last two for none nullable
```

array:

1. `comments: [Comment]`
2. `comments: [Comment!]`
3. `comments: [Comment]!`
4. `comments: [Comment!]!`

Let's try to explain the differences between the options. We'll also describe the Typescript return
value of each option (which will be automatically generated in the later Codegen chapter).

> Option 1, `[Comment]`, means that the elements of the array can are nullable and the array itself is

Reviewer 2: ```suggestion
Option 1, `[Comment]`, means that the elements of the array are nullable and the array itself is
```

nullable. The Typescript return type of that resolver would be `Array<Comment | null> | null`.
Representing `there's no Comment related to the Link` in that option can be applied in many

> different ways - an empty array, a null as array or an array with empty or null elements. That means

Reviewer 2: ```suggestion
different ways - an empty array, a null as array or an array with null elements. That means
```

that the client who consumes this will need to handle many cases.

Option 2, `[Comment!]`, means that the elements of the array are none nullable, but the array can

> return empty or null. The Typescript return type of that resolver would be `Array<Comment> | null`.

Reviewer 2: ```suggestion
be empty or null. The Typescript return type of that resolver would be `Array<Comment> | null`.
```
